spec/models/student_loans/permitted_parameters_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe StudentLoans::PermittedParameters do
+  let(:claim) { create(:tslr_claim) }
+  let(:permitted_parameters) { described_class.new(claim) }
+
+  describe "params" do
+    context "when no responses have come from verify" do
+      it "includes all parameters" do
+        expect(permitted_parameters.params).to eq([
+          :qts_award_year,
+          :claim_school_id,
+          :employment_status,
+          :current_school_id,
+          :mostly_teaching_eligible_subjects,
+          :full_name,
+          :address_line_1,
+          :address_line_2,
+          :address_line_3,
+          :address_line_4,
+          :postcode,
+          :date_of_birth,
+          :gender,
+          :teacher_reference_number,
+          :national_insurance_number,
+          :has_student_loan,
+          :student_loan_country,
+          :student_loan_courses,
+          :student_loan_start_date,
+          :student_loan_repayment_amount,
+          :email_address,
+          :bank_sort_code,
+          :bank_account_number,
+          TslrClaim::SUBJECT_FIELDS,
+          eligibility_attributes: [:qts_award_year],
+        ])
+      end
+    end
+
+    context "when a field has come from verify" do
+      before do
+        claim.verified_fields = ["gender"]
+      end
+
+      it "skips that field" do
+        expect(permitted_parameters.params).to eq([
```

Comment:
As above, it would be good if we could make this spec more robust. e.g. `StudentLoans::PermittedParameters::PARAMETERS - [:gender]`